### ISSUE TITLE
Create a job for building --without-ssl

### ISSUE THREAD

**Issue by danbev** [nodejs/build#1574]
I would be great to have a job that builds and tests node configuring it using the `--without-ssl` flag. Currently it is easy to forget to test this option and PRs are merged without us noticing.

I noticed https://github.com/nodejs/build/issues/643 exists and perhaps that should be reopened instead. Feel free to close this if that is this the case.

**Comment by refack** [comment]
Thanks for the issues.
I'll follow up.

**Comment by refack** [comment]
So it seems @rvagg already [wrote the script for this](https://ci.nodejs.org/job/node-test-commit-linux-containered/jobConfigHistory/showDiffFiles?timestamp1=2018-04-10_05-52-32&timestamp2=2018-04-23_08-33-42), it just wasn't turned on (probably because the test suit failed)

**Comment by richardlau** [comment]
@refack https://github.com/nodejs/build/issues/643#issuecomment-383580308

**Comment by refack** [comment]
Job is now ready, and will pass once https://github.com/nodejs/node/pull/24376 land: https://ci.nodejs.org/job/refack-node-test-commit-linux-containered/15/

**Comment by danbev** [comment]
> Job is now ready, and will pass once [nodejs/node#24376](https://github.com/nodejs/node/pull/24376) land: https://ci.nodejs.org/job/refack-node-test-commit-linux-containered/15/

Awesome, thanks!

**Comment by rvagg** [comment]
I honestly don't know why that was never enabled (just requires adding a label to the job iirc), perhaps it was due to too many failures without openssl.

**Comment by refack** [comment]
I can imagine it was because the test suit was failing, and in several places, not just tests, for example calls to npm don't work. But now  nodejs/node#24376 should be the last bloker.

**Comment by refack** [comment]
With https://github.com/nodejs/node/pull/24376 landed 3 days ago, PRs should be either fresh, or "rebasable", so IMHO we're ready to enable.
Sanity on `master` https://ci.nodejs.org/job/refack-node-test-commit-linux-containered/16/

**Comment by refack** [comment]
Oooof, new test fails - [`test.parallel/test-cli-node-print-help`](https://ci.nodejs.org/job/refack-node-test-commit-linux-containered/16/nodes=ubuntu1604_sharedlibs_withoutssl_x64/testReport/junit/(root)/test/parallel_test_cli_node_print_help/)

**Comment by danbev** [comment]
> Oooof, new test fails - [`test.parallel/test-cli-node-print-help`](https://ci.nodejs.org/job/refack-

I just ran into this. I can take a look at if and create a PR for it if you like?

**Comment by refack** [comment]
> I just ran into this. I can take a look at if and create a PR for it if you like?

I thought about doing it myself, but be my guest :)

**Comment by rvagg** [comment]
VersionSelectorScript.groovy:

```groovy
  [ /sharedlibs_withoutssl/,          anyType,     lt(10)  ],
```

So that's allowing it to run on 10+. Should we be switching that to `lt(12)` since that change isn't backported?

**Comment by richardlau** [comment]
Having this job would allow us to catch things like https://github.com/nodejs/node/pull/26182 (most likely due to a recent eslint update).

**Comment by refack** [comment]
Turned it on https://ci.nodejs.org/job/node-test-commit-linux-containered/nodes=ubuntu1604_sharedlibs_withoutssl_x64/ but blocked by https://github.com/nodejs/node/pull/26182

If someone could ping me when it's solved, I'll turn it back on.

**Comment by danbev** [comment]
@refack We have merged https://github.com/nodejs/node/pull/26182 now. Would you be able to turn the job back on? Thanks!

**Comment by richardlau** [comment]
Might be worth holding off until the security releases are done just to be on the safe side: https://github.com/nodejs/build/issues/1699

**Comment by richardlau** [comment]
Security releases are done. Ping @refack I think we can turn this on now.

**Comment by refack** [comment]
Turned on. Let's see https://ci.nodejs.org/job/node-test-commit-linux-containered/10929/

**Comment by refack** [comment]
https://ci.nodejs.org/job/node-test-commit-linux-containered/10929/nodes=ubuntu1604_sharedlibs_withoutssl_x64/ is ✔️ .
Closing.

**Comment by danbev** [comment]
Awesome, thanks @refack, @rvagg, and @richardlau!

**Comment by rvagg** [comment]
failing on v10.x with test-cli-node-print-help still https://ci.nodejs.org/view/All/job/node-test-commit-linux-containered/nodes=ubuntu1604_sharedlibs_withoutssl_x64/10933/

fix needs to be backported or we need to update jenkins/scripts/VersionSelectorScript.groovy to do `lt(11)` rather than `lt(10)`

**Comment by danbev** [comment]
I can take a look at backporting but it won't be until Monday at the earliest I'm afraid.

**Comment by rvagg** [comment]
@danbev ok, so let's leave the VersionSelectorScript with `lt(10)` for now anticipating getting it changed early next week. But @BethGriggs will need to be aware of the failing test (test-cli-node-print-help on node-test-commit-linux-containered / withoutssl) that can be ignored while preparing for 10.15.3 in the meantime.

**Comment by danbev** [comment]
@rvagg It looks like the fix for this is already on [v10.x-staging](94e3560cb4). I'm probably missing something obvious here (I'm a little sleep deprived and home with a sick kid today).

**Comment by richardlau** [comment]
> 
> 
> @rvagg It looks like the fix for this is already on [v10.x-staging](94e3560cb4). I'm probably missing something obvious here (I'm a little sleep deprived and home with a sick kid today).

The build Rod referenced was against v10.15.2. The fix was included in the v10.15.3 release that went out yesterday.

**Comment by richardlau** [comment]
There is, however, a failure on `v11.x` that we're not seeing on neither `master` nor `v10.x`: https://github.com/nodejs/node/issues/26480
